src/transaction/log_compress.c

```diff
@@ -191,16 +224,17 @@ log_diff (LOG_ZIP_SIZE_T undo_length, const void *undo_data, LOG_ZIP_SIZE_T redo
  * log_zip_alloc - allocate LOG_ZIP structure
  *   return: LOG_ZIP structure or NULL if error
  *   length(in): log_zip data buffer to be allocated
- *   is_zip(in): to be used zip or not
  *
  * Note:
  */
 LOG_ZIP *
-log_zip_alloc (LOG_ZIP_SIZE_T size, bool is_zip)
+log_zip_alloc (LOG_ZIP_SIZE_T size)
 {
   LOG_ZIP *log_zip = NULL;
   LOG_ZIP_SIZE_T buf_size = 0;
 
+  assert (size <= LZ4_MAX_INPUT_SIZE);
```

Comment:
it might compromise backward compatibility.

Reply:
From checking the log_zi_alloc usage I see size argument is upper-bounded by IO_MAX_PAGE_SIZE (16K). I think this assert is safe from the backward compatibility point of view.

Reply:
Ok, I understood.